src/doc/src/reference/registries.md

```diff
@@ -336,10 +335,11 @@ responses have a JSON object with the following structure:
 }
 ```
 
-Servers may also respond with a 404 response code to indicate the requested
-resource is not found (for example, an unknown crate name). However, using a
-200 response with an `errors` object allows a registry to provide a more
-detailed error message if desired.
+If the response has this structure Cargo will display the detailed message to the user, even if the response code is 200.
+If the response code indicates an error and the content does not have this structure, Cargo will display to the user a
+ message intended to help debugging the server error. A server returning an `errors` object allows a registry to provide a more
+detailed or user centric error message. The first release of Cargo to support registries Cargo 1.34 (released 2019-04-11) will only 
+show a human readable error if the code is 200, this has been fixed in all versions since Cargo 1.35 (released 2019-05-23).
```

Comment:
I don't think this historical information needs to be included. This was a tiny window 3 years ago.  Also, there is usually a hyphen before `centric` when combined with another word.

```suggestion
detailed or user-centric error message.
```